src/handlers/handle_config.py

```diff
@@ -5,105 +5,109 @@
-        FTP_Port: 21
-  # If a different folder should be used to store the updated plugins change to (True/False) and the path below
-        SeperateDownloadPath : False
-        PathToSeperateDownloadPath: /plugins/updated
-  # Change the path below if the plugin folder path is different on the SFTP/FTP server (Change only if you know what you are doing)
-        PluginFolderOnServer: /plugins
-    """
-	# load ruamel.yaml to get the # commands right in the yaml code
-	yaml = ruamel.yaml.YAML()
-	code = yaml.load(configuration)
-	with open("pluGET_config.yaml", "w") as config_file:
-		yaml.dump(code, config_file)
-
-	config_file_path = os.path.abspath("pluGET_config.yaml")
+	shutil.copyfile('src/config-sample.yaml', 'config.yaml')
+	config_file_path = os.path.abspath("config.yaml")
 	print(f"Path of config file: {config_file_path}")
 	print("Config created. Edit config before executing again!")
 	input("Press any key + enter to exit...")
 	sys.exit()
 
-
 def validate_config() -> None:
 	"""
-	Validates the config variables after config class is loaded and exit if error is detected and print error
+	Check for missing entries in the config
 	"""
-	accepted_values = [
-		("local", "sftp", "ftp")
-	]
-	# exit afterwards if there is an error in config
-	exit_afterwards = False
-	config = config_value()
-	# rich console for nice colors
+	yaml = ruamel.yaml.YAML()
+	with open("config.yaml", "r") as config_file:
+		data = yaml.load(config_file)
 	console = Console()
-	if config.connection not in accepted_values[0]:
-		console.print(f"Error in Config! Accepted values for key 'Connection' are {accepted_values[0]}",
-		style="bright_red")
+	exit_afterwards = False
+
+	if "Servers" not in data:
+		console.print(f"Config file is malformed",style="bright_red",highlight=False)
 		exit_afterwards = True
-	if exit_afterwards:
+	else:
+		for server in data["Servers"]:
+			temp = data["Servers"][server]
+			keys = ["Connection","ServerRootPath","PluginFolderPath","SeperateDownloadPath"]
+			remote_keys = ["Server","Username","Password","port"]
+			
+			if temp == None:
+				console.print(f"'{str(server)}' is malformed",style="bright_red",highlight=False)
+				exit_afterwards = True
+			else:
+				for cur in keys:
+					if cur not in temp:
+						console.print(f"'{str(server)}' is missing key '{cur}'",style="bright_red",highlight=False)
+						exit_afterwards = True
+
+				if "Connection" in temp:
+					accepted_values = ("local", "sftp", "ftp")
+					if temp["Connection"] not in accepted_values:
+						console.print(f"'{server}' has invalid key 'Connection' Valid options: {accepted_values}",style="bright_red",highlight=False)
+						exit_afterwards = True
+
+					if temp["Connection"] == "sftp" or temp["Connection"] == "ftp":
+						for cur in remote_keys:
+							if cur not in temp:
+								console.print(f"'{str(server)}' is missing key '{cur}'",style="bright_red",highlight=False)
+								exit_afterwards = True
+	if exit_afterwards: 
 		sys.exit()
```

Comment:
Replace all `exit_afterwards` with `sys.exit()` to just exit on the first error and remove all `if ... else` structures to make it easier to read. I can do that too if you don't know exactly what I mean